### PR TITLE
Kustomization openapi ConfigMapGenerator and SecretGenerator

### PR DESCRIPTION
**Pull request by phanimarupaka**
@pwittrock @monopole 

This PR adds kustomization openapi definitions for ConfigMapGenerator and SecretGenerator.

### CODE REVIEW

File: kyaml/openapi/kustomizationSwagger.json [hunk 1]
```diff
@@ -0,0 +1,442 @@
+{
+  "definitions": {
+    "ConfigMapArgs": {
```
Comment (pwittrock) [inline kyaml/openapi/kustomizationSwagger.json:3]: The Kubernetes types have a prefix that includes the api group.  We should do something similar.

File: kyaml/openapi/kustomizationSwagger.json [hunk 2]
```diff
@@ -0,0 +1,442 @@
+{
+  "definitions": {
```
Comment (pwittrock) [inline kyaml/openapi/kustomizationSwagger.json:2]: This should include a definition for the Kustomization `Kind`

File: kyaml/openapi/kustomizationSwagger.json [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+        },
+        "commonAnnotations": {
+          "patternProperties": {
+            ".*": {
+              "type": "string"
+            }
+          },
+          "type": "object"
+        },
+        "patchesStrategicMerge": {
```
Comment (pwittrock) [inline kyaml/openapi/kustomizationSwagger.json:146]: define the x-kubernetes extensions for arrays that should be merged.

**Comment by pwittrock** [comment]
/ok-to-test

Did you generate this with openapi-gen or hand craft it?

File: kyaml/openapi/kustomizationSwagger.json (hunk 1 above)
Comment (phanimarupaka) [reply kyaml/openapi/kustomizationSwagger.json]: Done.

File: kyaml/openapi/kustomizationSwagger.json (hunk 2 above)
Comment (phanimarupaka) [reply kyaml/openapi/kustomizationSwagger.json]: Added

File: kyaml/openapi/kustomizationSwagger.json (hunk 3 above)
Comment (phanimarupaka) [reply kyaml/openapi/kustomizationSwagger.json]: Done.

**Comment by pwittrock** [comment]
Thanks!  Are these generated from the kustomize types?  I didn't see any changes to the kustomize structs.

**Comment by phanimarupaka** [comment]
Yes. They are generated from kustomize types by using a different tool and made some manual changes to look like kube-api-gen. I am following up on generating it with kube-api-gen. Since the setup will be done with this PR, the next PRs will be around updating swagger.json/go and adding more tests.

File: kyaml/openapi/kustomizationapi/swagger.json [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      "type": "object",
+      "x-kubernetes-group-version-kind": [
+        {
+          "group": "kustomize.config.k8s.io",
+          "kind": "ConfigMapArgs",
+          "version": "v1beta1"
+        }
+      ]
+    },
+    "io.k8s.api.apps.v1.SecretArgs": {
```
Comment (phanimarupaka) [inline kyaml/openapi/kustomizationapi/swagger.json:19]: should this start with kustomize.config.k8s.io.v1beta1 instead ?

**Comment by pwittrock** [comment]
SG, please follow up with generating this through the kubernetes openapi gen tool.

**Pull request merged**